console/console-init/ui/src/Pages/AddressSpace/AddressesListWithFilterAndPaginationPage.tsx

```diff
@@ -55,12 +92,13 @@ function AddressesListFunction() {
           {totalAddresses > 0 && (
             <Pagination
               itemCount={totalAddresses}
-              perPage={10}
-              page={1}
-              onSetPage={() => {}}
-              widgetId="pagination-options-menu-top"
-              onPerPageSelect={() => {}}
-              style={{ paddingTop: 16, paddingBottom: 16 }}
+              perPage={perPage}
+              page={page}
+              onSetPage={(_: any, page: number) => handlePageChange(page)}
```

Comment:
closures should never be passed as properties like this, because they'll force the component to re-render since the function instance will be different between renders. That's the reason why we use `useCallback`. `handlePageChange` should then be updated to accept the two args required by `onSetPage`.

The same will have to be done for all the closures down here.